src/ucs/config/parser.c

```diff
@@ -1185,27 +1198,22 @@ ucs_status_t ucs_config_parser_fill_opts(void *opts, ucs_config_field_t *fields,
         goto err;
     }
 
-    /* Find if env_prefix consists of base prefix and custom prefix, base prefix
+    /* Find if env_prefix consists of base prefix and sub prefix, base prefix
      * should not contain underscores "_" except at the very end
-     * e.g in env_prefix = "LONG_PREFIX_UCX_LOG_LEVEL":
-     * "LONG_PREFIX_" is custom_prefix
+     * e.g in env_prefix = "LONG_PREFIX_UCX_":
+     * "LONG_PREFIX_" is sub prefix
      * "UCX_"         is base prefix */
-    full_prefix_len = strlen(env_prefix);
-    cstm_prefix_len = full_prefix_len - 2;
-    while ((cstm_prefix_len >= 0) && (env_prefix[cstm_prefix_len] != '_')) {
-        cstm_prefix_len -= 1;
-    }
-    cstm_prefix_len += 1;
+    base_prefix = ucs_config_parser_get_base_prefix(env_prefix);
 
     /* Apply environment variables */
-    status = ucs_config_apply_env_vars(opts, fields, env_prefix + cstm_prefix_len,
-                                       table_prefix, 1, ignore_errors);
+    status = ucs_config_apply_env_vars(opts, fields, base_prefix, table_prefix,
+                                       1, ignore_errors);
     if (status != UCS_OK) {
         goto err_free;
     }
 
     /* Apply environment variables with custom prefix */
-    if (cstm_prefix_len != 0) {
+    if (base_prefix != env_prefix) {
```

Comment:
instead of checking here, need to check if base_prefix !=NULL before using it in line 1209.
ucs_config_parser_get_base_prefix() would return NULL if there is no base prefix

Reply:
If this still relevant after sub_prefix_len += 1 fix to  ucs_config_parser_get_base_prefix?

Reply:
seems more reasonable to return NULL if sub prefix does not exist